crates/slumber_core/src/template.rs

```diff
@@ -947,33 +947,49 @@ mod tests {
     }
 
     #[rstest]
-    #[case::no_chains(vec!["foo!", "bar!"], Some("bar!"), "bar!")]
+    #[case::no_chains(vec!["foo!", "bar!"], vec!["foo!", "bar!"])]
+    #[case::chains(vec!["foo!", "{{chains.command}}"], vec!["foo!", "command_output"])]
```

Comment:
Rather than each test case iterating through each option, I think it'd be better to take an `index` param defined in each test case for which option we want to select, and `expected` should remain a `&str`. I like the idea of being thorough and checking them all, but I also prefer to keep test logic as simple as possible. In my experience, loops are a big contributor to test complexity and can make it hard to debug stuff. Then I would just add two test cases for the chains, `case::chains_0` to select `foo!` and `case::chains_1` to select the chain.

Reply:
ah, sounds good! Happy to simplify here.